source/tls_channel_handler.c

```diff
@@ -130,8 +130,63 @@ int aws_tls_ctx_options_init_client_mtls_from_path(
 
     return AWS_OP_SUCCESS;
 }
+#    if defined(__APPLE__)
+int aws_tls_ctx_options_init_client_mtls_custom_keychain(
+    struct aws_tls_ctx_options *options,
+    struct aws_allocator *allocator,
+    const struct aws_byte_cursor *cert,
+    const struct aws_byte_cursor *pkey,
+    const char *keychain_path) {
+    AWS_ZERO_STRUCT(*options);
+    options->minimum_tls_version = AWS_IO_TLS_VER_SYS_DEFAULTS;
+    options->cipher_pref = AWS_IO_TLS_CIPHER_PREF_SYSTEM_DEFAULT;
+    options->verify_peer = true;
+    options->allocator = allocator;
+    options->max_fragment_size = g_aws_channel_max_fragment_size;
+    options->keychain_path = keychain_path;
```

Comment:
I mentioned before, but this is a foot gun. copy it to an aws_string inside this function.

Reply:
I'll change both init functions to allocate that with `aws_string_new_from_c_str(options->allocator, keychain_path)` and free it with `aws_string_destroy()` in `aws_tls_ctx_options_clean_up()`